UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs

```diff
@@ -1187,5 +1192,82 @@ private void ColourTile(MatrixInfo matrixInfo, Vector3Int localPos, LayerType la
 		}
 
 		#endregion
+
+		#region DebugCommands
+
+		private static IEnumerator KillLights()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<LightSource>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
+				stationObject.Integrity.ForceDestroy();
+				currentIndex++;
+			}
+		}
+
+		private static IEnumerator SelfPowerEverything()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<APCPoweredDevice>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
+				stationObject.ChangeToSelfPowered();
+				currentIndex++;
+			}
+		}
+
+		private static IEnumerator TurnOnAllEmergancyLights()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<LightSource>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
+				stationObject.ServerChangeLightState(LightMountState.Emergency);
+				currentIndex++;
+			}
+		}
+
+
+		[Command(requiresAuthority = false)]
+		public void DestroyAllLights()
+		{
+			if(IsAdmin() == false) return;
+			PlayerManager.LocalPlayerScript.StartCoroutine(KillLights());
+			Chat.AddSystemMsgToChat(
+				"<color=blue>Lights are being destroyed to save energy and spice up the crew-members' working experience.</color>",
+				MatrixManager.MainStationMatrix);
+		}
+
+		[Command(requiresAuthority = false)]
+		public void SelfSuficeAllMachines()
+		{
+			if(IsAdmin() == false) return;
+			PlayerManager.LocalPlayerScript.StartCoroutine(SelfPowerEverything());
+			Chat.AddSystemMsgToChat(
+				"<color=blue>An admin is updating all machines on the station to not require APCs.</color>",
+				MatrixManager.MainStationMatrix);
+		}
+
+		[Command(requiresAuthority = false)]
+		public void TurnOnEmergencyLightsStationWide()
+		{
+			if(IsAdmin() == false) return;
+			PlayerManager.LocalPlayerScript.StartCoroutine(TurnOnAllEmergancyLights());
+			Chat.AddSystemMsgToChat(
+				"<color=red>Emergency Lights active.</color>",
+				MatrixManager.MainStationMatrix);
+		}
```

Comment:
You can't use that IsAdmin() like that serverside, do the same as the other Commands do

Reply:
![image](https://user-images.githubusercontent.com/34368774/204271881-1df21863-4c75-4336-b557-d4a8005b39ce.png)

Just to double-check, are you aware about this change and if so; what is wrong with it?
I see no issues with it at all.

Reply:
ah, i see the issue now

Reply:
Done